### DISCUSSION TITLE
How to deploy efficiently my local Docusaurus v3 blog to my FTP server

### DISCUSSION THREAD

**Discussion by cavo789** [facebook/docusaurus#9711] [Q&A]
Hi all 

On my machine, I'm hosting my local blog and, right now, I'm running `yarn build` on my machine to get content in my `build` folder. When done, I start my FTP client (using an automation script) and synchronize the `build` local folder with my `/var/www/html` folder on my FTP server so, in short, I copy new files from build to my FTP.

It works but certainly not the best way to do I think. Or I miss something important.

The problem of `yarn build` is: it's slow and change timestamp every time.

With the ever-increasing number of items, yarn build gets slower and slower; more and more files have to be generated and since the timestamps change each time, synchronization with the FTP server doesn't work and everything always has to be resent from my PC to my server.

There's something I'm missing, that's for sure.

Do you have any clues? How do you publish your local blog to an FTP server to feed your online site?

**Thanks**

**Comment by slorber** [comment]
> more and more files have to be generated and since the timestamps change each time, synchronization with the FTP server doesn't work

Modern CDNs to not rely on timestamp to deploy new files: they compare file hashes instead of using last modified date.

Using FTP is not a great way to deploy your site.

Try Vercel or Netlify instead, it will likely be free and is super easy to setup. You just push to git and it deploys.

**Reply by cavo789** [reply]
Vercel and Netlify are companies no ? I'll take a look... but I'm already hosted since years elsewhere and don't want to change.  

Do you know of a workflow where I can edit my articles locally, create new ones, ... and just upload my blog to my github repo.

Ideal would be that github will make the deployment automagically:

* either the build is done on github and only the static, new, files are sent to my host
* or github makes a call (from a hook) and the build is sent to my web host
* or something else 

I've already done a lot of searching and I don't think I've found the best approach right now.

At the moment, I manage my blog locally, I do a `yarn build` locally and each time I copy all the files by FTP (via a script).  This way of doing things is obviously bad ;) 

Thanks

**Reply by slorber** [reply]
Again Vercel or Netlify solve this well. 

If you have done a lot of research, what you probably need now is to trust me and act. 

Just give these services 5 minute of your time and you'll be immediately convinced.

**Comment by unleashit** [comment]
Agree, Vercel, Netlify and I would add Cloudflare Pages to the list of good ways to deploy static content. All of them have simple integration with github to deploy on push (to master/main and/or specified branches).

That said, I understand wanting to self host and/or not wanting to use certain companies because of data privacy or cost. Personally, unless your server doesn't support ssh, I'd go with rsync (either locally or with a Github Action). But if you really want to keep using FTP, it seems there is actually a GH action for it:

https://github.com/marketplace/actions/ftp-deploy

**Reply by cavo789** [reply]
Hi all,

I've take time this morning to look at the ftp-deploy github action and the official doc https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions and it is working fine.

Thanks for @slorber and @unleashit !

Among other things, now I can update my blog from anywhere (also from github.dev) and the deployment will be made automatically.
